packages/flutter/lib/src/material/bottom_navigation_bar.dart

```diff
@@ -257,12 +263,12 @@ class _BottomNavigationTile extends StatelessWidget {
   Widget _buildFixedLabel() {
     return Align(
       alignment: Alignment.bottomCenter,
-      heightFactor: 1.0,
+      heightFactor: this.heightFactor,
       child: Container(
-        margin: const EdgeInsets.only(bottom: _kBottomMargin),
+        margin: const EdgeInsets.all(0.0),
         child: DefaultTextStyle.merge(
           style: TextStyle(
-            fontSize: _kActiveFontSize,
+            fontSize: item.showLabel ? _kActiveFontSize : 0.0,
```

Comment:
Instead of using font size to keep the text from being seen, wrap it in an `Opacity` widget that is either 1 or 0 depending on `.showLabel`. Set `.alwaysIncludesSemantics` to true on the [Opacity](https://docs.flutter.io/flutter/widgets/Opacity-class.html).

Reply:
@willlarche This changes will make unnecessary space at the bottom when label isn't visible.
<img width="117" alt="screen shot 2019-01-16 at 2 38 15 pm" src="https://user-images.githubusercontent.com/27461460/51228581-7b8de480-199c-11e9-888c-c3db7e583b0d.png">

Every `TabBarItem` would look like above if you just wrap that with the opacity. Would this be wanted?

Reply:
If you wrap the entire returned value of `_buildShiftingLabel` and `_buildFixedLabel` in an opacity there will be no extra space allotted to the text. In the shifting case the label does move up slightly but I believe that is intended:

<img width="300" src="https://user-images.githubusercontent.com/8975114/51291829-6d9c8b80-19bd-11e9-8351-dc8b5974fdc9.png">


<img width="300" src="https://user-images.githubusercontent.com/8975114/51291919-b9e7cb80-19bd-11e9-8c7f-d778f82dc557.png">

Reply:
I've just made the changes and pushed commits. Thanks.